weather.py

```diff
@@ -0,0 +1,306 @@
+
+        elif mode == "c":
+            max_pads = []
+            min_pads = []
+            max_pad = ""
+            min_pad = ""
+
+            max_temp = Weatherman.c_result["max_temp"]
+            min_temp = Weatherman.c_result["min_temp"]
+
+            # Run loop for all the days logged
+            for i in range(len(max_temp)):
+
+                # Making '+' Strings to pad to output
+                for j in range(max_temp[i]):
+                    max_pad += "+"
+
+                for j in range(abs(min_temp[i])):
+                    min_pad += "+"
+
+                # Adding color to strings
+                max_pad = colored(max_pad, 'red')
+                min_pad = colored(min_pad, 'blue')
+
+                # Adding results to a list to return to print
+                max_pads.append(max_pad)
+                min_pads.append(min_pad)
+
+                # Resetting values for next iteration
+                max_pad = ""
+                min_pad = ""
+
+            return max_temp, min_temp, max_pads, min_pads
+
+    # Method to Print report of calculated results
+    def Print_report(self, mode):
+
+        # Checking which argument is true to print accordingly
+        if mode == "e":
+            high_temp, low_temp, high_month, low_month, high_year, low_year = self.getData("e")  # Data to print
+
+            # Making use of format strings to output results
+            print(
+            f"""Report Number {Weatherman.report_num}
+                Highest: {high_temp}C on {high_month} {high_year}
+                Lowest: {low_temp}C on {low_month} {low_year}
+                    """)
+            Weatherman.report_num += 1  # Recording report number to display multiple reports
+
+        elif mode == "a":
+            high_avg_temp, low_avg_temp, mean_avg_hum = self.getData("a")
+            print(
+            f"""Report Number {Weatherman.report_num}
+                Highest Average: {high_avg_temp}C
+                Lowest Average: {low_avg_temp}C
+                Average Mean Humidity: {mean_avg_hum}%
+                    """)
+            Weatherman.report_num += 1
+
+        elif mode == "c":
+
+            max_temp, min_temp, max_pads, min_pads = self.getData("c")
+
+            print(f'''Report Number {Weatherman.report_num}''')
+
+            # Run loop for all the days logged
+            for i in range(len(max_temp)):
+
+                # If !b, Print according to c argument, else print according to bonus task
+                if self.__b == 0:
+                    print(
+                f"""        
+                {i+1} {max_pads[i]} {max_temp[i]}C
+                {i+1} {min_pads[i]} {min_temp[i]}C""")
+
+                else:
+                    print(
+                f"""
+                {i + 1} {min_pads[i]}{max_pads[i]} {min_temp[i]}C - {max_temp[i]}C""")
+
+            Weatherman.report_num += 1
+
+    # Method to read lines from the file and return
+    def Reader(self, mode):
```

Comment:
This method can be extracted as a separate module that is only responsible for reading a file and returns its content line by line. Reading complete files at once can cause system breakage, reading complete files means you are going to transfer complete file content into memory for smaller files it may not be an issue but when you will try to read larger files they will occupy all memory and system will become unusable. So, it's better to avoid this practice. As files are in CSV format you can also parse them and return a record at a time. Generators might be helpful to achieve this behavior. Validating input also shouldn't be the responsibility of this module.